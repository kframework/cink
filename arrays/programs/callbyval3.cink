int f(int a, int x) {
  if (x < 5) return x * 2;
  return x * 3;
}

int main() {
  int b[4];
  cout << f(b[1], 5);
  return 0;
}

