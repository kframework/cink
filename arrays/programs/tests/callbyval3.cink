int f(int a, int x) {
  if (a < 5) return x * 2;
  return x * 3;
}

int main() {
  int b[4];
  b[1] = 3;
  b[2] = b[1];
  cout << f(b[2], 5);
  return 0;
}

