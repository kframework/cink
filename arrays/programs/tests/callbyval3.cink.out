10<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            0
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            f |-> #symScalLoc(1)
            main |-> #symScalLoc(2)
        </env>
    </control>
    <store>
        seq-array ( 4 , #symLoc(3) ) |-> store ( store ( const-array ( 4 ,
           0 ) , 1 , 3 ) , 2 , 3 )
        #symScalLoc(1) |-> lambda ( int a, int x @ if ( (a < 5) ) return (x * 
          2) ; else { } return (x * 3) ; )
        #symScalLoc(2) |-> lambda (  @ int (b [ 4 ]) ; ((b [ 1 ]) = 3) ; ((b [
           2 ]) = (b [ 1 ])) ; (cout << (f ( ((b [ 2 ]) , 5) ))) ; return 0 ; 
          )
        #symScalLoc(4) |-> 3
        #symScalLoc(5) |-> 5
    </store>
    <genv>
        f |-> #symScalLoc(1)
        main |-> #symScalLoc(2)
    </genv>
</T>
